server/api/image.py

```diff
@@ -170,9 +199,10 @@ def metadataGenerator():
         Description('Download multiple images as a ZIP file.')
         .param('include', 'Which types of data to include.', required=False,
                enum=['all', 'images', 'metadata'], default='all')
-        .param('datasetId', 'The ID of the dataset to download.', required=False)
-        .param('filter', 'Filter the images by a PegJS-specified grammar (causing "datasetId" to '
-                         'be ignored).',
+        .param('filter', 'Filter the images by a PegJS-specified grammar.',
+               required=False)
+        .param('imageIds',
+               'A JSON array of up to 100 image IDs, which will be exclusively included.',
```

Comment:
One solution I've seen discussed to deal with a large number of query parameters is to use a POST request instead. Has this been considered?

Reply:
By my rough math, we could probably handle 300-500 imageIDs without too much of an impact (perhaps we'd have to just change [Nginx's `large_client_header_buffers` option](http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers). Beyond that, we'd also want to consider how such a large query was being sent to MongoDB, even if we could get to the server via POST or otherwise. Additionally, if the images are being hand selected via a web client, the client itself would need to track and display the selection state of many individual images, which does not scale well either.

So, I think having some limit on the number of images is reasonable in general. I think we could safely bump up the 100 number in the future, but I don't foresee this ever allowing truly large numbers of selections.

Accordingly, I think we can keep the purity of using a `GET` here for now. I'd be fine to revisit this if the scaling requirements change in the future.

Reply:
Thanks for explaining the rationale. This all makes sense to me.